### PR TITLE
Closes #580 | Add Dataloader VSoLSCSum

### LINKED ISSUE #580: Create dataset loader for VSoLSCSum
Dataloader name: `vsolscsum/vsolscsum.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?vsolscsum

|     Dataset| vsolscsum |
|-------------|---|
| Description | The Vietnamese dataset for social context summarization  The dataset contains 141 open-domain articles along with 3,760 sentences, 2,448 extracted standard sentences and comments as standard summaries and 6,926 comments in 12 events. This dataset was manually annotated by human. Note that the extracted standard summaries also include comments. |
| Subsets | - |
| Languages | vie |
| Tasks | Summarization |
| License | Creative Commons Attribution 4.0 (cc-by-4.0) |
| Homepage | https://github.com/nguyenlab/VSoLSCSum-Dataset |
| HF URL | - |
| Paper URL | https://aclanthology.org/W16-5405/ |

### PR DESCRIPTION
**Pull request by muhammadravi251001**
**Title**: Add Dataloader VSoLSCSum

**First line PR Message**: Closes https://github.com/SEACrowd/seacrowd-datahub/issues/580

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      Calzorari, Nicoletta  and
+      Choi, Key-Sun",
+    booktitle = "Proceedings of the 12th Workshop on {A}sian Language Resources ({ALR}12)",
+    month = dec,
+    year = "2016",
+    address = "Osaka, Japan",
+    publisher = "The COLING 2016 Organizing Committee",
+    url = "https://aclanthology.org/W16-5405",
+    pages = "38--48",
+    abstract = "This paper presents VSoLSCSum, a Vietnamese linked sentence-comment dataset, which was manually created to treat the lack of standard corpora for social context summarization in Vietnamese. The dataset was collected through the keywords of 141 Web documents in 12 special events, which were mentioned on Vietnamese Web pages. Social users were asked to involve in creating standard summaries and the label of each sentence or comment. The inter-agreement calculated by Cohen{'}s Kappa among raters after validating is 0.685. To illustrate the potential use of our dataset, a learning to rank method was trained by using a set of local and social features. Experimental results indicate that the summary model trained on our dataset outperforms state-of-the-art baselines in both ROUGE-1 and ROUGE-2 in social context summarization.",
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/vsolscsum/vsolscsum.py:65]: The abstract is also mentioned in the `_DESCRIPTION`, the description below class (line 110-127) and the beginning (line 16-33). Let's remove this line

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    each annotator makes a binary decision in order to indicate \
+    that whether this sentence is a summary candidate (YES) or not \
+    (NO). If three annotators agree yes, this sentences is labeled by 3. \
+    Therefore, the label of each sentence or comment ranges from 1 to 5\
+    (1: very poor, 2: poor, 3: fair, 4: good; 5: perfect). The standard \
+    summary sentences are those which receive at least three agreements \
+    from annotators. The inter-agreement calculated by Cohen's Kappa \
+    after validation among annotators is 0.685.
+"""
+
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/vsolscsum/vsolscsum.py:34]: Let's remove these lines.

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    voting among social annotators. For example, given a sentence, \
+    each annotator makes a binary decision in order to indicate \
+    that whether this sentence is a summary candidate (YES) or not \
+    (NO). If three annotators agree yes, this sentences is labeled by 3. \
+    Therefore, the label of each sentence or comment ranges from 1 to 5\
+    (1: very poor, 2: poor, 3: fair, 4: good; 5: perfect). The standard \
+    summary sentences are those which receive at least three agreements \
+    from annotators. The inter-agreement calculated by Cohen's Kappa \
+    after validation among annotators is 0.685.
+    """
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/vsolscsum/vsolscsum.py:127]: This is the same as `_DESCRIPTION`, so would you please write a more simplified description other than the one in `_DESCRIPTION`?

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                    "text_2": str(row["document_and_comment"]),
+                    "text_1_name": "summary",
+                    "text_2_name": "document_and_comment",
+                }
+
+            yield index, example
+
+
+# This template is based on the following template from the datasets package:
+# https://github.com/huggingface/datasets/blob/master/templates/new_dataset_script.py
```
Comment (jen-santoso) [inline seacrowd/sea_datasets/vsolscsum/vsolscsum.py:244]: Let's remove this line

**Review by jen-santoso — CHANGES_REQUESTED** [review]
Hi @muhammadravi251001 
Thank you for implementing the dataloader, it runs well! 
Just some nits regarding the description.
Also, just to make sure, run `make check_file` command one more time before pushing the update, and it's good to go.

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py (hunk 2 above)
Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vsolscsum/vsolscsum.py:34]: I just follow the instructions on `/templates/template.py` here:

https://github.com/SEACrowd/seacrowd-datahub/blob/94bb4a18f345ef773a6922ce119005a74f0bdaa1/templates/template.py#L17-L29

Specifically on this line:

https://github.com/SEACrowd/seacrowd-datahub/blob/94bb4a18f345ef773a6922ce119005a74f0bdaa1/templates/template.py#L29

Do you still recommend removing those line, Sir?

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py (hunk 1 above)
Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vsolscsum/vsolscsum.py]: Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/609/commits/18815e7622ee0d853fd0d903a3366a26d9015b86.

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py (hunk 4 above)
Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vsolscsum/vsolscsum.py]: Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/609/commits/60c52750e9f190d1254e40b107e0235072619df0.

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py (hunk 3 above)
Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vsolscsum/vsolscsum.py]: Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/609/commits/ccad08500fcf3ab456aea35a2c8ab7b734404d4f.

File: seacrowd/sea_datasets/vsolscsum/vsolscsum.py (hunk 2 above)
Comment (jen-santoso) [reply seacrowd/sea_datasets/vsolscsum/vsolscsum.py:34]: As to unify the format with the rest of the submitted and approved dataloaders, yes.

Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vsolscsum/vsolscsum.py]: Alright, Miss. It's already solved on https://github.com/SEACrowd/seacrowd-datahub/pull/609/commits/c4a2bfd076a0449b07453c253e89bf6b7e04d480. Thanks!

**Review by jen-santoso — APPROVED** [review]

**Review by luckysusanto — APPROVED** [review]
Manual test looks good, code is clean!
Thanks for the code ^^

**Comment by muhammadravi251001** [comment]
> Manual test looks good, code is clean! Thanks for the code ^^

Thanks for the approval, Lucky!

**Comment by jen-santoso** [comment]
As both reviewers have approved, will merge this directly to master.
@luckysusanto don't hesitate to merge by the "squash and merge" method when two reviewers approved and no conflicts found.

**Pull request merged**